File: src/array/lc4.java
package array;

public class lc4 {
	public static double findMedianSortedArrays(int[] nums1, int[] nums2) {
		int n1 = nums1.length;
		int n2 = nums2.length;
		int n = n1 + n2;
		int mergeVector[] = new int[n];
		int index = 0;
		int index1 = 0;
		int index2 = 0;

		while (index <= n / 2 && index1 < n1 && index2 < n2) {
			if (nums1[index1] < nums2[index2]) {
				mergeVector[index] = nums1[index1];
				index1++;
			} else {
				mergeVector[index] = nums2[index2];
				index2++;
			}
			index++;
		}
		while (index <= n / 2 && index1 < n1) {
			mergeVector[index] = nums1[index1];
			index1++;
			index++;
		}

		while (index <= n / 2 && index2 < n2) {
			mergeVector[index] = nums2[index2];
			index2++;
			index++;
		}

		if (n % 2 == 0) {
			return ((double) (mergeVector[n / 2] + mergeVector[n / 2 - 1])) / 2;
		} else {
			return mergeVector[n / 2];
		}

	}

	public static void main(String[] args) {
		int x1[] = { 1, 3 }, x2[] = { 2 };
		double res = findMedianSortedArrays(x1, x2);
		System.out.println(res);
	}
}

File: c++/array/lc4.cpp
# include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
		int size1 = nums1.size();
		int size2 = nums2.size();
		vector<int> nums(size1+size2);
		int size = size1+ size2;
        //建立变量i1和i2分别表示遍历数组nums1和nums2的下标，i表示数组nums下标
        int i1 = 0, i2 = 0, i = 0;

        //循环直到i1和i2均遍历到数组末尾
        while (i1 < size1 || i2 < size2) {
            //nums1首元素大于nums2或i2已遍历到末尾
            if (i2 >= size2 || (i1 < size1 && nums1[i1] <= nums2[i2])) {  
                nums[i++] = nums1[i1++];
            }
            //nums2首元素大于nums1或i1已遍历到末尾
            else {
                nums[i++] = nums2[i2++];
            }
        }
		return (size1+size2) % 2 == 0 ? ((double)nums[size / 2 - 1] + (double)nums[size / 2]) / 2 : (double)nums[size/2];
    }
};

int main(){
	vector<int> num1 = {1,3,5,9};
		vector<int> num2 = {3,8,9};

	Solution().findMedianSortedArrays(num1,num2);
	return 0;
}
